src/cpp/lpnamer/ActiveLinks.cpp

```diff
@@ -1,2 +1,121 @@
 
 #include "ActiveLinks.h"
+#include "LinkProfileReader.h"
+#include "Candidate.h"
+#include <unordered_set>
+
+
+void ActiveLinks::addCandidate(const CandidateData& candidate_data, const std::map<std::string, LinkProfile>& profile_map){
+
+    int indexLink = getIndexOf(candidate_data.link_id);
+
+    if (indexLink == -1)
+    {
+        ActiveLink link(candidate_data);
+        _links.push_back(link);
+        indexLink = _links.size() - 1;
+    }
+
+    _links[indexLink].addCandidate(candidate_data, profile_map);
+}
+
+void ActiveLinks::addCandidates(const std::vector<CandidateData>& candidateList, const std::map<std::string, LinkProfile>& profile_map)
+{
+    std::unordered_set<std::string> setCandidatesNames;
```

Comment:
this is a good idea, it can be an private field of the class to be used as a service information to properly build the vector of active_links.

**We can rename `ActiveLinks` into `ActiveLinksBuilder`**.
Its responsibility is to build a vector of `ActiveLinks`.
Then in order to use the information we simply pass a `const vector<ActiveLinks>`as parameter to the functions that will need the information.